crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -51,8 +53,23 @@ impl ProtocolFee {
             }
         }
 
-        let protocol_fees = match &self.policy {
-            policy::Policy::Surplus(variant) => variant.apply(&order, quote),
+        let order_ = boundary::Amounts {
+            sell: order.data.sell_amount,
```

Comment:
This is becoming a bit noisy. Maybe we can have an implementation like `impl From<&Order> for boudnary::Amounts` and the same for `Quote`.
Then we could do
```rust
boundary::is_order_outside_market_price(order.into(), quote.into())
```

Reply:
For documentation purposes: We tried out this idea in person and it turned out to not work. The issue is Rust's orphan rules that states for you to implement a trait for a type your crate must define the trait or the type. Since `Order` got re-exported from a crate the Order type is considered foreign so we are not allowed to implement `From` here.